src/jit/flowgraph.cpp

```diff
@@ -1001,6 +1001,78 @@ flowList* Compiler::fgSpliceOutPred(BasicBlock* block, BasicBlock* blockPred)
 }
 
 //------------------------------------------------------------------------
+// fgExpensiveSortedPredCheck: Assert that the pred list for all Basic Blocks is sorted
+//
+// Assumptions:
+//    -- This only make sense to check when with the full predecessor lists, not the cheap preds lists.
+//
+// The eventual goal is to assert this is true. Until we can assume that this is
+// true, just return whether it is sorted or not.
+
+bool Compiler::fgExpensiveSortedPredCheck()
+{
+    assert(!fgCheapPredsValid);
+
+#ifdef DEBUG
+
+    if (JitConfig.JitSlowDebugChecksEnabled() != 0)
+    {
+
+        bool sorted = true;
+
+        for (BasicBlock* block = fgFirstBB; block != nullptr; block = block->bbNext)
+        {
+            return fgExpensiveSortedPredCheck(block);
```

Comment:
Needs to be `if (!fgExpensiveSortedPredCheck(block)) return false;`.

Currently, it only checks the first block.